src/3_branch.js

```diff
@@ -301,8 +310,28 @@ Branch.prototype['init'] = wrap(callback_params.CALLBACK_ERR_DATA, function(done
 		done(err, data && utils.whiteListSessionData(data));
 	};
 
+	var allTheWaysToSayHidden =
+		[{ 'hidden': 'visibilitychange' },
+		{ 'mozHidden': 'mozvisibilitychange' },
+		{ 'msHidden': 'msvisibilitychange' },
+		{ 'webkitHidden': 'webkitvisibilitychange' }],
+
+	attachVisibilityEvent = function(data) {
+		for(var index = 0; index < allTheWaysToSayHidden.length; index++) {
+				var hidden = Object.keys(allTheWaysToSayHidden[index])[0],
+				event = allTheWaysToSayHidden[index][hidden];
+			if (typeof document[hidden] !== 'undefined') {
+				document['addEventListener'](event, function() {
+					if (!document[hidden]) { check_has_app(data, finishInit); }
+				}, false);
+				break;
+			}
+		}
+	};
+
 	if (WEB_BUILD && sessionData  && sessionData['session_id'] && (utils.processReferringLink(link_identifier) === sessionData['referring_link'] || link_identifier === sessionData['click_id'])) {
-		finishInit(null, sessionData);
+		attachVisibilityEvent(sessionData);
+		check_has_app(sessionData, finishInit);
```

Comment:
This is actually the case where we don't call init, because we already have a session, so we do want to call it here. Where we do call init at L#381, I'm just calling attachVisibilityEvent.